applications/MappingApplication/custom_utilities/mapper_utilities.cpp

```diff
@@ -51,6 +51,26 @@ void AssignInterfaceEquationIds(Communicator& rModelPartCommunicator)
     rModelPartCommunicator.SynchronizeNonHistoricalVariable(INTERFACE_EQUATION_ID);
 }
 
+void AssignInterfaceEquationIdsOnConditions(Communicator& rModelPartCommunicator)
```

Comment:
Ideally you would generalize the function that works on nodes, maybe by passing a range, or iterator start and end